### PR TITLE
ci: Fix change detection (part 3)

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
@alyssawilk the idea with this Pr is that we remove all the requirements on checks from branch protection and replace them with just the `complete` job - that way the other jobs can either run or be skipped without failing the pipeline

File: .azure-pipelines/pipelines.yml [hunk 1]
```diff
@@ -451,6 +451,19 @@ stages:
...
       condition: always()
 
+  - job: complete
+    displayName: "linux_x64"
+    dependsOn: ["bazel", "coverage"]
+    pool:
+      vmImage: "ubuntu-20.04"
+    condition: |
+      and(
+        in(dependencies.bazel.result, 'Succeeded', 'SucceededWithIssues', 'Skipped'),
```
Comment (alyssawilk) [inline .azure-pipelines/pipelines.yml:461]: Can we comment about this job and how it works inline?  I would assume that SucceededWithIssues would be bad and not grouped with the other two.
/wait

Comment (phlax) [reply .azure-pipelines/pipelines.yml:461]: i think we need `SucceededWithIssues` - not sure exactly what it does but this condition is functionally equivalent to using the `succeeded()` queue time var

that is my interpretation of:

`succeeded()` `==` `in(variables['Agent.JobStatus'], 'Succeeded', 'SucceededWithIssues')`

here https://learn.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#succeeded

Comment (phlax) [reply .azure-pipelines/pipelines.yml:465]: comment added

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**